pkg/endpoint/identifiers.go

```diff
@@ -46,34 +43,29 @@ func (e *Endpoint) HumanStringLocked() string {
 // GetK8sNamespaceAndPodName returns the corresponding namespace and pod
 // name for this endpoint.
 func (e *Endpoint) GetK8sNamespaceAndPodName() string {
-	e.unconditionalRLock()
-	defer e.runlock()
-
 	return e.getK8sNamespaceAndPodName()
 }
 
 func (e *Endpoint) getK8sNamespaceAndPodName() string {
 	return e.K8sNamespace + "/" + e.K8sPodName
 }
 
-// SetK8sPodName modifies the endpoint's pod name
-func (e *Endpoint) SetK8sPodName(name string) {
-	e.unconditionalLock()
+// SetK8sPodNameLocked modifies the endpoint's pod name
+// Only used for tests.
+func (e *Endpoint) SetK8sPodNameLocked(name string) {
```

Comment:
Would be better to move non-production code to tests, or better yet eliminate the need for this in the tests by using production code for creating the endpoints also in tests, if possible?

Reply:
I agree completely; unfortunately the tests that reference this are out-of-package. I was trying to make this PR as minimal as possible.

The tests that use this functionality could _probably_ be refactored to stop, I can check.

Reply:
There, I managed to remove some of the silly setter methods.